applications/FluidDynamicsApplication/custom_utilities/mass_conservation_utility.cpp

```diff
@@ -246,43 +302,112 @@ void MassConservationCheckProcess::ComputeVolumesAndInterface( double& positiveV
+        if (geom_is_cut){
+            // element is cut by the surface (splitting)
+            Kratos::unique_ptr<ModifiedShapeFunctions> p_modified_sh_func = nullptr;
+            Vector w_gauss_interface(3, 0.0);
 
-        // instead of using data.isCut()
-        for (unsigned int pt = 0; pt < rGeom.Points().size(); pt++){
-            if ( rGeom[pt].FastGetSolutionStepValue(DISTANCE) > 0.0 ){
-                pt_count_pos++;
-            } else {
-                pt_count_neg++;
+            Vector Distance( r_geom.PointsNumber(), 0.0 );
+            for (unsigned int i = 0; i < r_geom.PointsNumber(); i++){
+                // Control mechanism to avoid 0.0 ( is necessary because "distance_modification" possibly not yet executed )
+                if ( r_geom[i].FastGetSolutionStepValue(DISTANCE) == 0.0 ){
+                    it_elem->GetGeometry().GetPoint(i).FastGetSolutionStepValue(DISTANCE) = -1.0e-7;
+                }
+                Distance[i] = r_geom[i].FastGetSolutionStepValue(DISTANCE);
             }
-        }
 
-        if ( pt_count_pos == rGeom.PointsNumber() || pt_count_neg == rGeom.PointsNumber() ){
-            // all nodes are on one side and the element is not split
-            continue;
+            if ( r_geom.PointsNumber() == 3 ){ p_modified_sh_func = Kratos::make_unique<Triangle2D3ModifiedShapeFunctions>(it_elem->pGetGeometry(), Distance); }
+            else if ( r_geom.PointsNumber() == 4 ){ p_modified_sh_func = Kratos::make_unique<Tetrahedra3D4ModifiedShapeFunctions>(it_elem->pGetGeometry(), Distance); }
+            else { KRATOS_ERROR << "The utility can not be applied on this kind of element" << std::endl; }
+
+            // Concerning their area, the positive and negative side of the interface are equal
+            p_modified_sh_func->ComputeInterfacePositiveSideShapeFunctionsAndGradientsValues(
+                    shape_functions,                    // N
+                    shape_derivatives,                  // DN
+                    w_gauss_interface,                  // includes the weights of the GAUSS points (!!!)
+                    GeometryData::GI_GAUSS_2);          // second order Gauss integration
+
+            // negative side outwards area normal vector values for the Gauss pts. of given quadrature
+            std::vector<Vector> normal_vectors;
+            p_modified_sh_func->ComputeNegativeSideInterfaceAreaNormals(
+                normal_vectors,
+                GeometryData::GI_GAUSS_2
+            );
+
+            // iteration over all 3 or 6 integration points
+            for ( unsigned int i_gauss = 0; i_gauss < w_gauss_interface.size(); i_gauss++)
+            {
+                const double& r_weight = w_gauss_interface[i_gauss];
+                const auto& r_N = row(shape_functions, i_gauss);
+
+                auto& r_normal = normal_vectors[i_gauss];
+                r_normal /= norm_2( r_normal );
```

Comment:
I'd check that the norm of `r_normal` is not close to 0.0.

Reply:
added